### PR TITLE
OnStep: Rate Compensation

### PR DESCRIPTION
**Pull request by azwing**
few changes and fixes

### CODE REVIEW

**Comment by knro** [comment]
Ok it's 71 commits, can be please squash the commits and resubmit?

**Comment by azwing** [comment]
I am lost.
Do not understand how to squash and by trying I finally succeeded to destroy everything and lost all the work.
So better to stop here and try to restart from scratch.

Sorry for the inconvenience

**Comment by TallFurryMan** [comment]
Do not despair. Most probably your work is still in your local repository. Check the result of the command "git reflog", that will show the last manipulations you did, and you should be able to roll back to a known state.

**Comment by TallFurryMan** [comment]
You can also restore your work from this PR branch itself as last resort.

**Comment by knro** [comment]
@TallFurryMan any idea how to re-run the CI tests here? or it needs a commit?

@azwing Do not worry about squash, I found out I could do it here.

**Comment by azwing** [comment]
Ok, thanks for support.

I will try this and may be I will find out how this squash works :-)

File: libindi/drivers/telescope/lx200_OnStep.cpp [hunk 1]
```diff
@@ -99,14 +99,18 @@ bool LX200_OnStep::initProperties()
 
     // ============== MOTION_CONTROL_TAB
 
-    IUFillNumber(&MaxSlewRateN[0], "maxSlew", "Rate", "%g", 1.0, 9.0, 1.0, 5.0);    //2.0, 9.0, 1.0, 9.0
+    IUFillNumber(&MaxSlewRateN[0], "maxSlew", "Rate", "%g", 0.0, 9.0, 1.0, 5.0);    //2.0, 9.0, 1.0, 9.0
```
Comment (knro) [inline libindi/drivers/telescope/lx200_OnStep.cpp]: use "%.f " for format since this is integer and not double.

**Comment by knro** [comment]
No need to squash, just resubmit the PR again.

**Comment by TallFurryMan** [comment]
@knro there is a way to retrig the tests from the github interface for the repository owner - you in that case. But pay attention to the effective tests: the CircleCI config only runs the master branch, so if the PR doesn't source from the master branch, no tests are run actually.

**Comment by TallFurryMan** [comment]
@azwing to squash your PR with no damage to your history, use a development branch. When you are ready to push the PR, merge your development branch to your master branch as explained at https://stackoverflow.com/questions/5308816/how-to-use-git-merge-squash/5309051#5309051

**Comment by knro** [comment]
@TallFurryMan I was about to ask you that actually! no sure how to re-run the tests.

@azwing there is NO need to squash, the option is now available in Github when I merge.. just 1 click!

File: libindi/drivers/telescope/lx200_OnStep.cpp (hunk 1 above)
Comment (azwing) [reply libindi/drivers/telescope/lx200_OnStep.cpp]: Thanks Jasem,

I made the changes. Will re-issue a PR.
In the meantime I saw this magic button "squash and merge"

**Pull request closed without merging**